lib/core/src/lib/common/utils/date-fns-utils.ts

```diff
@@ -15,12 +15,13 @@
  * limitations under the License.
  */
 
-import {ar, cs, da, de, enUS, es, fi, fr, it, ja, nb, nl, pl, ptBR, ru, sv, zhCN} from 'date-fns/locale';
+import { format, parse } from 'date-fns';
+import { ar, cs, da, de, enUS, es, fi, fr, it, ja, nb, nl, pl, ptBR, ru, sv, zhCN } from 'date-fns/locale';
 
 export class DateFnsUtils {
     static getLocaleFromString(locale: string): Locale {
         let dateFnsLocale: Locale;
-        switch(locale) {
+        switch (locale) {
```

Comment:
there is a bug here, you are not checking the locales correctly, as there might be "en-US", "en-GB", etc, see https://developer.mozilla.org/en-US/docs/Web/API/Navigator/languages

Reply:
you might want to lowercase the input and split by the dash, to get the first part of the locale

Reply:
https://date-fns.org/v2.30.0/docs/I18n

Reply:
we need supporting the following locales by default: https://github.com/Alfresco/alfresco-ng2-components/blob/98c0a3c7bef00e6069a279ef47a01d2732784ee0/lib/core/src/lib/language-menu/service/language.service.ts#L26